src/client/components/pages/Courses/MeetingModal.tsx

```diff
@@ -227,20 +248,66 @@ const MeetingModal: FunctionComponent<MeetingModalProps> = function ({
     if (isVisible) {
       setSaveError('');
       setMeetingModalFocus();
-      setAllMeetings(instanceMeetings);
+      if (!initialized) {
+        setAllMeetings(instanceMeetings);
+        setInitialized(true);
+      }
     } else {
       setCurrentEditMeeting(null);
       setShowRoomsData(null);
+      setInitialized(false);
     }
+    /**
+     * Checks to see if there are any unsaved changes in the modal when the user
+     * refreshes the page. If there are unsaved changes, the browser displays a
+     * warning message to confirm the page reload. If the user selects cancel, the
+     * user can continue making changes in the modal.
+     */
+    const onBeforeUnload = (event: Event) => {
+      if (!isChanged) return;
+      event.preventDefault();
+      // It's unclear whether TS will account for browser incompatibility here,
+      // so we use all three methods of setting the confirmation message,
+      // as detailed here:
+      // https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event
+      const legacyEvent = (event as unknown as LegacyEvent);
+      legacyEvent.returnValue = confirmMessage;
+      return confirmMessage;
```

Comment:
I'm still annoyed that Chrome doesn't follow the spec here, but I'll get over it.

One thing that is probably worth changing is that, instead of creating a `LegacyEvent` interface just for this use case, what if we set the `returnValue` to `false` (which is the original type) and return the `confirmMessage` separately. That still seems to work on my end:

```ts
// Need to disable this rule for browser compatibility reasons
// eslint-disable-next-line no-param-reassign
event.returnValue = false;
return confirmMessage;
```

Reply:
Thanks for the suggestion! I updated this in a594352.